avocado/utils/configure_network.py

```diff
@@ -10,193 +10,388 @@
-                log.error("unable to get peer interface: %s", ex)
+            _run_command(cmd, self.remote_session)
+        except Exception as ex:
+            raise NWException("Failed to ping: {}".format(ex))
+
+    def save(self, ipaddr, netmask):
+        """Save current interface IP Address to the system configuration file.
+
+        If the ipaddr is valid (currently being used by the interface)
+        this will try to save the current settings into /etc/. This
+        check is necessary to avoid inconsistency. Before save, you
+        should add_ipaddr, first.
+
+        Currently, only RHEL, Fedora and SuSE are supported. And this
+        will create a backup file of your current configuration if
+        found.
+
+        :param ipaddr : IP Address which need to configure for interface
+        :param netmask: Network mask which is associated to the provided IP
+        """
+        current_distro = distro.detect()
+
+        filename = "ifcfg-{}".format(self.name)
+        if current_distro.name in ['rhel', 'fedora']:
+            path = "/etc/sysconfig/network-scripts"
+        elif current_distro.name == 'SuSE':
+            path = "/etc/sysconfig/network"
         else:
-            return peer_interface
+            msg = 'Distro not supported by API. Could not save ipaddr.'
+            raise NWException(msg)
+
+        if ipaddr not in self.get_ipaddrs():
+            msg = ('ipaddr not configured on interface. To avoid '
+                   'inconsistency, please add the ipaddr first.')
+            raise NWException(msg)
 
+        self._write_to_file("{}/{}".format(path, filename),
+                            {'TYPE': self.if_type,
+                             'BOOTPROTO': 'none',
+                             'NAME': self.name,
+                             'DEVICE': self.name,
+                             'ONBOOT': 'yes',
+                             'IPADDR': ipaddr,
+                             'NETMASK': netmask,
+                             'IPV6INIT': 'yes',
+                             'IPV6_AUTOCONF': 'yes',
+                             'IPV6_DEFROUTE': 'yes'})
 
-def is_interface_link_up(interface):
+    def set_mtu(self, mtu, timeout=30):
+        """Set a new MTU value to this interface.
+
+        This method will try to set a new MTU value to this interface,
+        if fails it will raise a NWException. Also it will wait until
+        the Interface is up before returning or until timeout be
+        reached.
+
+        You must have sudo permissions to run this method on a host.
+
+        :param mtu:  mtu size that need to be set
+        :param timeout: how many seconds to wait until the interface is
+                        up again. Default is 30.
+        """
+        cmd = "ip link set %s mtu %s" % (self.name, mtu)
+        _run_command(cmd, self.remote_session, sudo=True)
+        wait.wait_for(self.is_link_up, timeout=timeout)
+        if mtu != self.get_mtu():
```

Comment:
@beraldoleal self.get_mtu() this one is on int type, we are passing mtu value as a string,so here its failing.

Reply:
@beraldoleal  and need the return value True or false here.

Reply:
@smruti77, I will add a `int()` to fix this. But since that MTU values are numbers I will keep returning integers here.

About the return value, to keep consistency with the other methods, we are doing the action and raising an Exception in case of failures. Users should catch them.